docs/guides.md

```diff
@@ -24,7 +24,6 @@ guides/custom_model.md
 guides/run_maxtext_via_xpk.md
 guides/run_maxtext_via_pathways.md
 guides/data_input_pipeline.md
-guides/data_input_perf.md
```

Comment:
Does this file not exist or something? Just wondering why removing this is needed for the build to pass

Reply:
This file was removed when this guide was split into four separate pages, in https://github.com/AI-Hypercomputer/maxtext/pull/2244/files